azkaban-common/src/main/java/azkaban/executor/AbstractExecutorManagerAdapter.java

```diff
@@ -273,14 +273,19 @@ public String submitExecutableFlow(final ExecutableFlow exflow, final String use
     }
   }
 
+  @Override
+  public DispatchMethod getDispatchMethod(final ExecutableFlow flow) {
+    return getDispatchMethod();
+  }
+
   protected String uploadExecutableFlow(
       final ExecutableFlow exflow, final String userId, final String flowId,
       String message) throws ExecutorManagerException {
     final int projectId = exflow.getProjectId();
     exflow.setSubmitUser(userId);
     exflow.setStatus(getStartStatus());
     exflow.setSubmitTime(System.currentTimeMillis());
-    exflow.setDispatchMethod(getDispatchMethod());
+    exflow.setDispatchMethod(getDispatchMethod(exflow));
```

Comment:
So when Flow is submitted/uploaded to the table, we determine the dispatchMethod and put it as part of ExecutableFlow so that it will be added as a column in exeuction_flows table and also becomes part of the flow_data.  So every query which will be used from there will keep dispatch_method into account.